### PR TITLE
test: Update the new ZTEST APIs tests/net/traffic_class

### PR DESCRIPTION
**Pull request by hunterhu3000**
TEST=twister -T tests/net/traffic_class

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

File: tests/net/traffic_class/src/main.c [hunk 1]
```diff
@@ -968,46 +969,42 @@ static void test_traffic_class_recv_data_mix_all_2(void)
...
+	test_traffic_class_recv_data_prio_ee();
+	test_traffic_class_recv_data_prio_ca();
+	test_traffic_class_recv_data_prio_vi();
+	test_traffic_class_recv_data_prio_vo();
+	test_traffic_class_recv_data_prio_ic();
+	test_traffic_class_recv_data_prio_nc();
+	test_traffic_class_recv_data_mix();
+	test_traffic_class_recv_data_mix_all_1();
+	test_traffic_class_recv_data_mix_all_2();
+	test_traffic_class_cleanup_rx();
```
Comment (yperess) [inline tests/net/traffic_class/src/main.c:1007]: Are these all inter-dependent? Each test should be hermetic and run individually in any order.

**Review by yperess — CHANGES_REQUESTED** [review]

File: tests/net/traffic_class/src/main.c (hunk 1 above)
Comment (hunterhu3000) [reply tests/net/traffic_class/src/main.c]: These supposed to be independent, but actually not.
I have modified some code and made them independent. Thanks to review.

**Comment by hunterhu3000** [comment]
CONFIG_ZTEST_SHUFFLE can be set to y on this test case

**Review by enjiamai — DISMISSED** [review]

**Review by yperess — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Comment by enjiamai** [comment]
HI, @hunterhu3000 , sorry, I press the wrong button, it should be "rebase", could you please rebase and force-push it again, thanks!

**Comment by hunterhu3000** [comment]
> HI, @hunterhu3000 , sorry, I press the wrong button, it should be "rebase", could you please rebase and force-push it again, thanks!

Done

**Review by enjiamai — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**